main/src/ca/uwaterloo/flix/language/ast/LoweredMoreAst.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2025 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.ast
+
+import ca.uwaterloo.flix.language.ast.Purity.Pure
+import ca.uwaterloo.flix.language.ast.shared.SymUse.{EffSymUse, OpSymUse}
+import ca.uwaterloo.flix.language.ast.shared.*
+
+object LoweredMoreAst {
+
+  val empty: Root = Root(Map.empty, Map.empty, Map.empty, Map.empty, Set.empty, List.empty, None, Set.empty, Map.empty)
+
+  case class Root(defs: Map[Symbol.DefnSym, Def],
+                  enums: Map[Symbol.EnumSym, Enum],
+                  structs: Map[Symbol.StructSym, Struct],
+                  effects: Map[Symbol.EffSym, Effect],
+                  types: Set[SimpleType],
+                  anonClasses: List[AnonClass],
+                  mainEntryPoint: Option[Symbol.DefnSym],
+                  entryPoints: Set[Symbol.DefnSym],
+                  sources: Map[Source, SourceLocation]) {
+
+    def getMain: Option[Def] = mainEntryPoint.map(defs(_))
+
+  }
+
+  /**
+    * pcPoints is initialized by [[ca.uwaterloo.flix.language.phase.Reducer]].
```

Comment:
Is this true? Should pcPoints then be removed from LoweredAst?

Reply:
Sorry - i misspoke. `pcPoints` only exist in `ReducedAst`. It cannot be removed in `ReducedAst` since `Reducer` has type `ReducedAst -> ReducedAst`. So it has to be in both `ReducedAst` and `JvmAst`